src/test/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/integration/ActiveVisitorByPrisonerIdAndBookerReferenceTest.kt

```diff
@@ -0,0 +1,147 @@
+package uk.gov.justice.digital.hmpps.prison.visitbooker.registry.integration
+
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpHeaders
+import org.springframework.test.web.reactive.server.WebTestClient
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.controller.ACTIVATE_BOOKER_PRISONER_VISITOR_CONTROLLER_PATH
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.dto.PermittedVisitorDto
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.model.entity.Booker
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.model.entity.PermittedPrisoner
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.model.entity.PermittedVisitor
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.model.repository.PermittedVisitorRepository
+
+@DisplayName("Activate booker prisoner visitor")
+class ActiveVisitorByPrisonerIdAndBookerReferenceTest : IntegrationTestBase() {
+
+  private lateinit var booker: Booker
+
+  private lateinit var prisoner: PermittedPrisoner
+
+  @Autowired
+  private lateinit var permittedVisitorRepository: PermittedVisitorRepository
+
+  private lateinit var visitor1: PermittedVisitor
+  private lateinit var visitor2: PermittedVisitor
+
+  @BeforeEach
+  internal fun setUp() {
+    booker = createBooker(oneLoginSub = "123", emailAddress = "test@example.com")
+
+    val prisoners = createAssociatedPrisoners(
+      booker,
+      listOf(
+        PermittedPrisonerTestObject("AB123456", true),
+        PermittedPrisonerTestObject("AB789012", true),
+      ),
+      visitors = listOf(),
+    )
+
+    prisoner = prisoners[0]
+
+    val visitors = createAssociatedPrisonersVisitors(
+      prisoner,
+      listOf(
+        PermittedVisitorTestObject(12, false),
+        PermittedVisitorTestObject(34, false),
+      ),
+    )
+    visitor1 = visitors[0]
+    visitor2 = visitors[1]
+  }
+
+  @Test
+  fun `when prisoner is activated for booker all data is persisted and returned correctly`() {
```

Comment:
should we add a test for deactivate as well?

Reply:
Oh yes was ment to do that